userspace/libsinsp/cri_settings.cpp

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+Copyright (C) 2024 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include "cri.hpp"
+
+namespace libsinsp
+{
+namespace cri
+{
+
+settings::settings():
+	m_cri_unix_socket_paths(),
+	m_cri_timeout(1000),
+	m_cri_size_timeout(10000),
+	m_cri_runtime_type(CT_CRI),
+	m_cri_unix_socket_path(),
+	m_cri_extra_queries(true)
+{ }
+
+settings::~settings()
+{ }
+
+settings* settings::s_instance = nullptr;
```

Comment:
Same feedback as other PR, do we need raw pointers? If possible smart pointers going forward in libs, we have bee burnt a few times.

Reply:
The problem here is that I didn't take the static local variable approach (like Meyer's Singleton) but, instead, I went for a more straightforward singleton with a static variable in the global scope. Do you know if the static local variable approach guarantees the uniqueness of the object in all the cases? 🤔

Reply:
Thanks for the additional comment @therealbobo let's collectively check what the best approach would be here. Maybe the other maintainers have preferences as well.

Reply:
Thought more about it and I would prefer a static `get()` function that returns a reference to a `std::unique_ptr` similar to the existing `sinsp_dns_manager`. Reasons being that we adopt a more consistent style throughout the code base and we discussed in previous core maintainers meetings that we won't accept raw pointers anymore going forward unless absolutely necessary, which I believe we don't have a strong case here for an exception. WDYT?